diff -rpu boost_1_66_0_old/boost/type_traits/detail/common_arithmetic_type.hpp boost_1_66_0/boost/type_traits/detail/common_arithmetic_type.hpp
--- boost_1_66_0_old/boost/type_traits/detail/common_arithmetic_type.hpp	2017-12-14 00:56:49.000000000 +0100
+++ boost_1_66_0/boost/type_traits/detail/common_arithmetic_type.hpp	2018-04-26 14:14:44.020680512 +0200
@@ -205,10 +205,11 @@ private:
 #endif
 
     static bool cond();
+    static const int selector = sizeof(select( cond()? T(): U() ));
 
 public:
 
-    typedef typename arithmetic_type< sizeof(select( cond()? T(): U() )) >::type type;
+    typedef typename arithmetic_type< selector >::type type;
 };
 
 } // namespace type_traits_detail
