--- apache-log4cxx-0.10.0/src/main/cpp/stringhelper.cpp2018-04-19 09:48:50.762552	436 +0200
+++ apache-log4cxx-0.10.0-old/src/main/cpp/stringhelper.cpp2	008-04-01 00:34:09.000000000 +0200
@@ -28,7 +28,6 @@
 #endif
 #include <log4cxx/private/log4cxx_private.h>
 #include <cctype>
-#include <cstdlib>
 #include <apr.h>
 
 
diff -rpu apache-log4cxx-0.10.0/src/main/include/log4cxx/helpers/simpledateformat.h apache-log4cxx-0.10.0-old/src/main/include/log4cxx/helpers/simpledateformat.h
--- apache-log4cxx-0.10.0/src/main/include/log4cxx/helpers/simpledateformat.h2018-04-19 09:4	8:19.010260662 +0200
+++ apache-log4cxx-0.10.0-old/src/main/include/log4cxx/helpers/simpledateformat.h2008-04-01 00:34:26.000000000 +0200
@@ -26,1	0 +26,11 @@
 
 
 #include <log4cxx/helpers/dateformat.h>
-#include <locale>
 #include <vector>
 #include <time.h>
 
+namespace std { class locale; }
+
 namespace log4cxx
 {
         namespace helpers
