--- setup.py	2019-05-29 20:07:01.000000000 +0200
+++ PyTables-3.5.2/setup.py	2019-06-26 10:03:53.765389146 +0200
@@ -40,6 +40,10 @@
         import cpuinfo
         cpu_info = cpuinfo.get_cpu_info()
         cpu_flags = cpu_info['flags']
+        if 'SPACK_TARGET_FLAGS' in os.environ:
+            cpu_flags = set(cpu_flags)
+            cpu_flags -= set(['avx2', 'sse2'])
+            cpu_flags.update(set(os.environ['SPACK_TARGET_FLAGS'].split()))
     except Exception as e:
         print('cpuinfo failed, assuming no CPU features:', e)
         cpu_flags = []
