diff -up /dev/null entrypoints-0.2.2/setup.py 
--- /dev/null   2016-08-18 08:57:28.760870159 +0200
+++ entrypoints-0.2.2/setup.py   2016-09-13 19:09:54.854292820 +0200
@@ -0,0 +1,26 @@
+#!/usr/bin/env python
+
+import os, re
+from distutils.core import setup
+
+with open(os.path.join(os.path.dirname(__file__), 'entrypoints.py'), 'r') as f:
+    for line in f:
+        version_match = re.search(r"__version__ = ['\"]([^'\"]*)['\"]", line)
+        if version_match:
+            version = version_match.group(1)
+            break
+    else:
+        raise Exception("couldn't find version number")
+
+os.mkdir('entrypoints')
+os.rename('entrypoints.py', 'entrypoints/__init__.py')
+
+setup(name='entrypoints',
+      version=version,
+      description='Discover and load entry points from installed packages.',
+      author='Thomas Kluyver',
+      author_email='thomas@kluyver.me.uk',
+      url='https://github.com/takluyver/entrypoints',
+      py_modules=['entrypoints.py'],
+      install_requires=['configparser>=3.5'],
+     )
