--- a/configure.orig	2019-08-12 09:01:53.943727093 +0200
+++ b/configure	2019-08-12 09:03:37.967785678 +0200
@@ -7713,7 +7713,7 @@
 
 	# Check gcc's version
 	case "$(${QMAKE_CONF_COMPILER} -dumpversion)" in
-	    4*)
+	    [4-9]*)
 		;;
 	    3.4*)
 		canBuildQtXmlPatterns="no"
@@ -7734,7 +7734,7 @@
     *-g++*)
 	# Check gcc's version
 	case "$(${QMAKE_CONF_COMPILER} -dumpversion)" in
-	    5*|4*|3.4*)
+	    [4-9]*|3.4*)
 		;;
             3.3*)
                 canBuildWebKit="no"

