diff -Nr -U5 qt-everywhere-opensource-src-4.8.7/src/plugins/bearer/corewlan/corewlan.pro qt-everywhere-opensource-src-4.8.7.corewlan-new-osx/src/plugins/bearer/corewlan/corewlan.pro
--- qt-everywhere-opensource-src-4.8.7/src/plugins/bearer/corewlan/corewlan.pro	2015-05-07 16:14:48.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.7/src/plugins/bearer/corewlan/corewlan.pro	2019-08-09 11:00:48.205113813 +0200
@@ -5,11 +5,11 @@
 LIBS += -framework Foundation -framework SystemConfiguration
 
 contains(QT_CONFIG, corewlan) {
     LIBS += -framework CoreWLAN
 
-    isEmpty(QMAKE_MAC_SDK)|contains(QMAKE_MAC_SDK, ".*OSX10\.[6789]\.sdk/?$") {
+    isEmpty(QMAKE_MAC_SDK)|contains(QMAKE_MAC_SDK, ".*MacOSX10\.([6789]|1[0123])\.sdk") {
          LIBS += -framework Security
     }
 }
 
 HEADERS += qcorewlanengine.h \
