diff -Naur setup.py newsetup.py
--- ./setup.py	2019-02-11 13:53:39.333331848 +0100
+++ ./setup.py	2019-02-11 13:53:59.620867789 +0100
@@ -16,25 +16,6 @@
     with open('requirements-{0}.txt'.format(extra), 'r') as fp:
         extras_require[extra] = fp.read()
 
-# spade specific
-is_64bit = sys.maxsize > 2 ** 32
-is_python3 = float(sys.version[0:3]) > 2.7
-
-if is_python3:
-    if is_64bit:
-        urlretrieve('http://www.borgelt.net/bin64/py3/fim.so',
-                    'elephant/spade_src/fim.so')
-    else:
-        urlretrieve('http://www.borgelt.net/bin32/py3/fim.so',
-                    'elephant/spade_src/fim.so')
-else:
-    if is_64bit:
-        urlretrieve('http://www.borgelt.net/bin64/py2/fim.so',
-                    'elephant/spade_src/fim.so')
-    else:
-        urlretrieve('http://www.borgelt.net/bin32/py2/fim.so',
-                    'elephant/spade_src/fim.so')
-
 setup(
     name="elephant",
     version='0.6.0',
