--- setup.py	2018-04-03 17:37:22.507062147 +0200
+++ PyTables-3.3.0/setup.py	2018-04-03 18:45:52.940675005 +0200
@@ -35,6 +35,11 @@
 import cpuinfo
 cpu_info = cpuinfo.get_cpu_info()
 
+if 'SPACK_TARGET_FLAGS' in os.environ:
+    cpu_flags = set(cpu_flags)
+    cpu_flags -= set(['avx2', 'sse2'])
+    cpu_flags.update(set(os.environ['SPACK_TARGET_FLAGS'].split()))
+
 # The name for the pkg-config utility
 PKG_CONFIG = 'pkg-config'

